training_data_loader.cpp

```diff
@@ -490,6 +490,57 @@ struct FeaturedBatchStream : Stream<StorageT>
 
 extern "C" {
 
+    EXPORT SparseBatch* get_sparse_batch_from_fens(
+        const char* feature_set_c,
+        int num_fens,
+        const char* const* fens,
+        int* scores,
+        int* plies,
+        int* results
+    )
+    {
+        std::vector<TrainingDataEntry> entries;
```

Comment:
Perhaps it is also a good idea to validate for repeated fens, and use std::set or something similar to avoid storing duplicates?

Reply:
Doesn't matter, and if anything it should be done on the caller side. This is a very simple function.